src/billing-reports/payment-refund-report.component.tsx

```diff
@@ -0,0 +1,396 @@
+      const allResults = await fetchAllRefundPaymentReport(formatToYMD(startDate), formatToYMD(endDate), collector);
+
+      exportToExcel(columns, allResults, getValue, 'payment-refund-report.xlsx');
+    } catch (error) {
+      console.error('Export failed:', error);
+      setErrorMessage(t('errorExportingExcel', 'Failed to export to Excel.'));
+    } finally {
+      setLoading(false);
+    }
+  };
+  const refundedItemsColumns = ['service_name', 'qty_paid', 'refund_qty', 'unit_price', 'refund_reason'];
+
+  const headerDisplayMap: Record<string, string> = {
+    refund_id: t('refundId', 'Refund Id'),
+    payment_id: t('paymentId', 'Payment Id'),
+    cashier_name: t('cashierName', 'Cashier Name'),
+    submitted_on: t('submittedOn', 'Submitted On'),
+    approvedby: t('approvedBy', 'Approved By'),
+    confirmed_by: t('confirmedBy', 'Confirmed By'),
+    service_name: t('serviceName', 'Service Name'),
+    qty_paid: t('qtyPaid', 'Quantity Paid'),
+    refund_qty: t('refundQty', 'Refund Quantity'),
+    unit_price: t('unitPrice', 'Unit Price'),
+    refund_reason: t('refundReason', 'Refund Reason'),
+  };
+
+  const handleSearch = async (filters: Filters, pageNum = 1, pageSize = 50) => {
+    setLoading(true);
+    setErrorMessage(null);
+    try {
+      const formattedStart = formatToYMD(filters.startDate);
+      const formattedEnd = formatToYMD(filters.endDate);
+
+      const { results, total } = await fetchRefundPaymentReport(
+        formattedStart,
+        formattedEnd,
+        filters.collector,
+        1,
+        10000,
+      );
+      if (results.length > 0) {
+        const columnNames = results[0].record.map((item) => item.column);
+        setColumns(columnNames);
+      }
+
+      setResults(results);
+      setTotalRecords(total);
+      setPage(pageNum);
+      setPageSize(pageSize);
+      setCurrentFilters(filters);
+    } catch (error) {
+      console.error('Error fetching report:', error);
+      setErrorMessage(t('errorFetchingReport', 'Failed to load report data.'));
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const goToPage = (newPage: number) => {
+    setPage(newPage);
+    paginateResults(allResults, newPage, pageSize);
+  };
+
+  const getPageSizes = () => [50];
+
+  const formatDateTime = (dateString: string | string[] | undefined) => {
+    if (!dateString) return '-';
+    const val = Array.isArray(dateString) ? dateString[0] : dateString;
+    return dayjs(val).isValid() ? dayjs(val).format('YYYY-MM-DD HH:mm') : formatValue(val);
+  };
+
+  const closeModal = () => {
+    setModalOpen(false);
+    setSelectedRecord(null);
+  };
+
+  const groupedByRefundId: { [key: string]: ReportRow[] } = {};
+
+  results.forEach((row) => {
+    const refundId = getValue(row.record, 'refund_id');
+    if (!groupedByRefundId[refundId]) {
+      groupedByRefundId[refundId] = [];
+    }
+    groupedByRefundId[refundId].push(row);
+  });
+
+  const rows = Object.entries(groupedByRefundId)
```

Comment:
In the future you'd maybe want to extract this mapping logic into a pure helper function outside the component for readability and testability maybe during a code cleanup but you can as well move this to a util file